### ISSUE TITLE
[BUG][2.2.54] "systemctl daemon-reload" is needed after setting waagent-network-setup.service

### ISSUE THREAD

**Issue by yuxisun1217** [Azure/WALinuxAgent#2224]
**Describe the bug: A clear and concise description of what the bug is.**
There's a warning message if show waagent-network-setup.service status:
```
Warning: The unit file, source configuration file or drop-ins of waagent-network-setup.service changed on disk. Run 'systemctl daemon-reload' to reload units.
```
Seems after setting /usr/lib/systemd/system/waagent-network-setup.service file, it needs to run "systemctl daemon-reload".

**Distro and WALinuxAgent details (please complete the following information):**
 - Distro and Version: RHEL-8.5
 - WALinuxAgent-2.2.49.2 running on redhat 8.5
Python: 3.6.8
Goal state agent: 2.2.54.2

**Comment by larohra** [comment]
Hey @yuxisun1217, thanks for reporting this. This is a harmless warning thrown when the GuestAgent sets up the network-setup.service unit file. This service is mainly needed to run on service startup, apart from that this is not needed. Due to this the waagent doesn't exclusively call `systemctl daemon-reload` on the system.

This behavior is by design.

Could you please share how you got this message? And if it is of any concern from an admin standpoint to you? I just want to understand the impact of this message. Thanks!

**Comment by yuxisun1217** [comment]
Hi @larohra ,

Thank you for your reply! I see this message by running "systemctl status waagent-network-setup.service". I'm not quite sure if there's any feature impact, but I just think this warning message will confuse the customer. And actually IMO it should execute "systemctl daemon-reload" after modifying the .service file.

**Comment by larohra** [comment]
Hey @yuxisun1217, there is no feature impact but I think you're right, it would definitely confuse customers. I'm going to do a fix for this and release it as part of the next release. I'll resolve this issue once the fix is merged.

Thanks for reporting this!

**Comment by yuxisun1217** [comment]
OK thank you so much @larohra !

**Comment by larohra** [comment]
Fixed in #2225
